src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/HttpConnectionPool.cs

```diff
@@ -305,12 +305,12 @@ private object SyncObj
         {
             get
             {
-                Debug.Assert(!Monitor.IsEntered(_http11Connections));
-                return _http11Connections;
+                Debug.Assert(!Monitor.IsEntered(_originAuthority));
+                return _originAuthority;
             }
         }
 
-        public bool HasSyncObjLock => Monitor.IsEntered(_http11Connections);
+        public bool HasSyncObjLock => Monitor.IsEntered(_originAuthority);
```

Comment:
```suggestion
        public bool HasSyncObjLock => Monitor.IsEntered(SyncObj);
```
?

Reply:
The `SyncObj` getter asserts that it's not already taken

Reply:
That sounds wrong. :)

Reply:
I found it odd at first, but it has helped me spot an error once.
Do you think it'd be better if we moved the asssrts to all callers (~25 places in the pool)?

Reply:
Doesn't need to be in this PR, but I do think it's undesirable that getting the object to lock on requires it not already be acquired. That's really up to the consumer. They might be wanting to assert, they might be fine with recursive acquisiton, they might want to use explicit Enter/Exit calls (in which case it would necessarily be acquired to call Exit), etc. I'm surprised actually that in this use every one of those ~25 sites requires it to be unaquired.

Reply:
I doubt anything bad would happen in most of those even if we did recur.
I see it more so just as increasing confidence that we're not accidentally creating cycles between methods while holding the lock.